calculator.rb

```diff
@@ -0,0 +1,207 @@
+
+    return verify_user_input($stdin.gets.chomp,true)
+
+  # handling operator values that don't pass
+  elsif !should_be_number && !verify_operator(user_input)
+    puts "Oops. That's not a math operator I recognize. Can you say it differently?"
+    puts "I can do addition, subtraction, multiplication, division, modulo, and"
+    puts "exponent calculations."
+
+    return verify_user_input($stdin.gets.chomp,false)
+  end
+
+  # simplifies user operator
+  if !should_be_number
+    user_input = verify_operator(user_input)
+  end
+
+  # returns qualified user_input! if the interpreter gets this far, the user_input has
+  # passed all the qualification tests! :)
+  return user_input
+end
+
+
+
+# verifies that a number value. takes one parameter.
+def verify_number(user_number)
+  # first, get rid of spaces & commas
+  while (user_number.include? " ") || (user_number.include? ",")
+    user_number.slice!(" ")
+    user_number.slice!(",")
+  end
+
+  # if ASCII would say the number is above '/' and below ':',
+  # then according to ASCII it's a number in a string! n_n
+  # I was sooo excited to discover this! :D
+  # awesome ASCII chart: http://www.bibase.com/images/ascii.gif
+  if (user_number > "/") && (user_number < ":")
+    # since it's a number, returns the user_number
+    return user_number
+
+  # if it's not an ASCII character between '/' and ':',
+  # then it's not a number in a string
+  else
+    return false
+  end
+end
+
+
+
+# this actually simplifies the operator on top of verifying it
+# it takes one argument and checks & simplifies it
+def verify_operator(user_operator)
+  # we only need the first three characters for these checks
+  user_operator = user_operator[0..2].downcase
+
+  # setting valid operators for each type of operation
+  operators_addition = { keywords: %w{+ add plu}, symbol: "+" }
+  operators_subtraction = { keywords: %w{- sub min}, symbol: "-" }
+  operators_multiplication = { keywords: %w{* x mul tim}, symbol: "*" }
+  operators_division = { keywords: %w{/ div}, symbol: "/" }
+  operators_exponent = { keywords: %w{^ pow exp}, symbol: "^" }
+  operators_modulo = { keywords: %w{% mod}, symbol: "%" }
+
+  # putting the operators together for looping action
+  valid_operators = [operators_addition, operators_subtraction,
+    operators_multiplication, operators_division, operators_exponent,
+    operators_modulo]
+
+  # iterates through operators, calling check_keywords on each, and returning
+  # the symbol for that operator if it passes the check. because it _returns_
+  # something, the loop & whole method will break / complete upon finding a
+  # match.
+  valid_operators.each do |operators_set|
+    if check_keywords(operators_set[:keywords], user_operator)
+      return operators_set[:symbol]
+    end
+  end
+
+  # returns false if no keyword matches are found, so verify_answer will call
+  # itself again
+  return false
+end
+
```

Comment:
I love what you've done here! You did a great job tightening up your repeated code and now its even easier if you wanted to add another operation!